src/Vacuum/Vacuum_math.jl

```diff
@@ -0,0 +1,728 @@
+#############################################################
+# Julia Numerical Utilities: Function Summary & Fortran Mapping
+#
+# Interpolation & Smoothing:
+#   - spline1d(x, y, xq)           : Cubic spline interpolation (spl1d1)
+#   - spline1d_deriv(x, y, xq)     : Cubic spline derivative (spl1d2)
+#   - lagrange1d(x, y, xq)         : Lagrange interpolation (lagp, lagpe4, lag)
+#   - smooth(y, w)                 : Moving average smoothing (smooth, smooth0)
+#   - shift(y, nshift)             : Periodic array shift (shft)
+#
+# Green's Function & Legendre Kernel:
+#   - green(xs, zs, xt, zt, n)     : Green's function (green)
```

Comment:
The VacuumFields.jl package (written by me mainly) has the `n=0` Green's function at least, and since it's so costly, we've optimized it pretty well. Eventually maybe we move your higher-n version in there.